docs/source/blueprints/BP000.rst

```diff
@@ -0,0 +1,336 @@
+the blueprint's basic notions. This document is a blueprint to specify how we
+should write blueprints from now on.
+
+Motivation
+**********
+
+Depending on the project size, having very well defined and structured
+documents about the architecture decisions seems like an overkill, but could
+help projects of any size, including Avocado, save time, make better decisions
+and improve the way we socialize those decisions.
+
+Today in the Avocado project we have the good practice to submit an RFC to
+our mailing list or to use/open a GitHub issue when we have a new idea. RFCs are a
+widespread way to disclose the architecture decisions, but they are just one
+part of a longer process. During the RFC phase, we argue in favor of a
+proposal, and we are mostly concerned about collecting feedback. After this
+phase, we could go one step forward and consolidate the discussion in
+Blueprints (sometimes called ADRs - Architecture Decision Records). This could
+be the next step so we could better socialize our decisions for future readers.
+A very well defined and structured document has some advantages over an RFC,
+but it is not intended to replace it, just be a later stage to follow from it.
+
+With blueprints, we could not only, but mainly:
+
+  * Create better documents for future members and future reference, when we
+    are trying to answer the following questions:
+
+a) *why* are you doing this? (the “problem” or the “motivation”);
+b) *what* are you proposing to solve the problem? (your solution)?
+c) And *how* are you going to implement the proposed solution?
+
+Depending on the type of your blueprint, the answer for the last question (c)
+could be written in pseudocode, general text or might even not be necessary
+(although desired) — more details on the section named Specification.
+
+When using RFCs as email threads, there are no sections or headers, each
+contributor will try to send the RFC without following formal sections and
+headers. RFCs, as we use today are just thread discussions and are not focused
+on future review/reading.
+
+  * Make sure that another peer will be able to implement a feature/requirement.
+
+Blueprints are not for you; they are for the community. If you know that a
+problem exists and know how to fix it, the most natural course of action would
+be to start coding this fix and submitting a Pull Request. While this is still
+valid for most of the cases, some important architectural changes must be
+discussed first to explain the “why”, “what” and “how” to keep everyone on the
+same page, avoid unnecessary coding, and most importantly: allow others to
+implement it in case you are not available.
+
+  * Improve the code review quality and time.
+
+Having a better understanding of the problem and the big picture is better for
+code review. It is harder to capture that from the Pull Request (PR)
+description and PR changes. Developers that are aware of the problem tend to
```

Comment:
Especially if we are resolving large issues through multiple PRs- blueprints will for sure be indispensable there.